joblib/parallel.py

```diff
@@ -1423,7 +1423,28 @@ def dispatch_one_batch(self, iterator):
                 n_jobs = self._cached_effective_n_jobs
                 big_batch_size = batch_size * n_jobs
 
-                islice = list(itertools.islice(iterator, big_batch_size))
+                try:
+                    islice = list(itertools.islice(iterator, big_batch_size))
+                except Exception as e:
```

Comment:
Should we try to catch KeyboardInterrupt ?

Maybe catch BaseException, except StopIteration ?

Reply:
Hum, not sure about how this would change the behavior here...

What happens on keyboard interrupt for now?
I am in favor of keeping this simple for this PR and we can improve this in a follow up PR if necessary?

Reply:
>  I am in favor of keeping this simple for this PR and we can improve this in a follow up PR if necessary?

Yes let's do that.

I thought KeyboardInterrupt will be caught by whatever thread is being executed at that time but I don't find clear explanation of that, it might only be the main thread. I don't recall hearing from issues with keyboardinterrupt being raised here anyway so my suggestion is probably off.